src/app/AppToolbarArea.tsx

```diff
@@ -76,8 +73,8 @@ function createSlotPortal<T extends HTMLElement = HTMLElement>(opts: {
 
   return {
     reactContext,
-    SlotComponent,
-    PortalComponent,
-    PortalProvider,
+    SlotComponent: memo(SlotComponent),
+    PortalComponent: memo(PortalComponent),
+    PortalProvider: memo(PortalProvider),
```

Comment:
I was in this area trying to see if this could be done without a portal. It can, using a state of `React.ReactNode`. Then you need another effect to update the state when children change, so it ended up causing more re-renders than just using a portal. These memos are safe though, why not.